src/backend/gl/src/lib.rs

```diff
@@ -272,17 +272,32 @@ impl PhysicalDevice {
             panic!("Error querying info: {:?}", err);
         }
 
-        // opengl has no way to discern device_type, so we can try to infer it from the vendor string
-        let vender_lower = vendor.to_lowercase();
-        let inferred_device_type = if vender_lower.contains("intel") {
-            hal::adapter::DeviceType::IntegratedGpu
-        } else if vender_lower.contains("nvidia")
-            || vender_lower.contains("ati")
-            || vender_lower.contains("amd")
+        // opengl has no way to discern device_type, so we can try to infer it from the renderer string
+        
+        let renderer_lower = renderer.to_lowercase();
+        let strings_that_imply_integrated = [
+            " xpress", // space here is on purpose so we don't match express
+            "radeon hd 4200",  
```

Comment:
any reason to not match just `"radeon hd "?

Reply:
yes, most radeon hd gpus are not integrated